crypto/mod.ts

```diff
@@ -0,0 +1,247 @@
+          // if it supports the algorithm.
+          return stdCrypto.subtle.digestSync(algorithm, bytes);
+        } else if ((data as Iterable<BufferSource>)[Symbol.iterator]) {
+          return stdCrypto.subtle.digestSync(
+            algorithm,
+            data as Iterable<BufferSource>,
+          );
+        } else if (
+          (data as AsyncIterable<BufferSource>)[Symbol.asyncIterator]
+        ) {
+          const context = new wasmCrypto.DigestContext(name);
+          for await (const chunk of data as AsyncIterable<BufferSource>) {
+            const chunkBytes = bufferSourceBytes(chunk)!;
+            context.update(chunkBytes);
+          }
+          return context.digestAndDrop(length);
+        } else {
+          throw new TypeError(
+            "data must be a BufferSource or [Async]Iterable<BufferSource>",
+          );
+        }
+      } else if (webCrypto.subtle?.digest) {
+        // (TypeScript type definitions prohibit this case.) If they're trying
+        // to call an algorithm we don't recognize, pass it along to WebCrypto
+        // in case it's a non-standard algorithm supported by the the runtime
+        // they're using.
+        return await webCrypto.subtle.digest(
+          algorithm,
+          data as unknown as Uint8Array,
+        );
+      } else {
+        throw new TypeError(`unsupported digest algorithm: ${algorithm}`);
+      }
+    },
+
+    digestSync(
+      algorithm: DigestAlgorithm,
+      data: BufferSource | Iterable<BufferSource>,
+    ): ArrayBuffer {
+      algorithm = normalizeAlgorithm(algorithm);
+
+      const bytes = bufferSourceBytes(data);
+
+      if (bytes) {
+        return wasmCrypto.digest(algorithm.name, bytes, undefined);
+      } else if ((data as Iterable<BufferSource>)[Symbol.iterator]) {
+        const context = new wasmCrypto.DigestContext(algorithm.name);
+        for (const chunk of data as Iterable<BufferSource>) {
+          const chunkBytes = bufferSourceBytes(chunk)!;
+          context.update(chunkBytes);
+        }
+        return context.digestAndDrop(algorithm.length);
+      } else {
+        throw new TypeError(
+          "data must be a BufferSource or Iterable<BufferSource>",
+        );
+      }
+    },
+  },
+});
+
+/** Digest algorithms supported by WebCrypto. */
+const webCryptoDigestAlgorithms = [
+  "SHA-384",
+  "SHA-256",
+  "SHA-512",
+  // insecure (length-extendable and collidable):
+  "SHA-1",
+] as const;
+
+type DigestAlgorithmName = WasmDigestAlgorithm;
+
+type DigestAlgorithmObject = {
+  name: DigestAlgorithmName;
+  length?: number;
+};
+
+type DigestAlgorithm = DigestAlgorithmName | DigestAlgorithmObject;
+
+const normalizeAlgorithm = (algorithm: DigestAlgorithm) =>
+  ((typeof algorithm === "string") ? { name: algorithm.toUpperCase() } : {
+    ...algorithm,
+    name: algorithm.name.toUpperCase(),
+  }) as DigestAlgorithmObject;
+
+export default stdCrypto;
```

Comment:
We shouldn't use `export default`. We only use it in `/node` for API compatibility reasons.